vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -112,7 +185,14 @@ func (c *InstallerController) createInstallerController(operatorSpec *operatorv1
 			} else {
 				glog.V(2).Infof("%q is in transition to %d, but has not made progress", currNodeState.NodeName, currNodeState.TargetDeploymentGeneration)
 			}
-			break
+
+			if !pendingNewDeployment {
+				break
+			}
+
+			// fall through to start the next deployment
```

Comment:
Falling through means you may update status again.  This means that you need to have the new operatorstatus as returned by `UpdateStatus`

Reply:
Compare the updated logic:
- the "target > current" if clause logic is as before
- no fallthrough anymore, but a break
- on resync the first node is found that is not ready to its reported current deployment id. It's used as the new start of the pending deployment id.

=> the "operator dies" TODO is resolved as the operator can pick up an unready static pod at any time.
=> the new deployment flow is uniform again.